HomeWork2/DmytroFenevHW2.cpp

```diff
@@ -0,0 +1,151 @@
+    std::cout << "\tTask 2 - Lucky ticket\n";
+    std::cout << "\tTask 3 - Reversed number\n";
+    std::cout << "\tTask 4 - Sum of odd number\n";
+    std::cout << "\tTask 5 - Best divisor\n";
+  
+    std::cout << "\tEnter number of Task N1, N2, N3, N4, N5\n";
+    std::cin >> task_choise;
+    int x, n, tmp;
+   
+    switch (task_choise) 
+    {
+    case FIRST_TASK:
+        n = 1; //число цифр
+        std::cout << "\tEnter an integer" << std::endl;
+        std::cin >> x;
+        //xp = x; //сохранение х после подсчета количества цифр
+        tmp = x;
+        while ((tmp /= 10) > 0)
+            n++;
+        std::cout << "\tNumber of digits=" << n << std::endl;
+        int sum;
+        sum = 0;
+        for (int ii = 0; ii < n; ii++) 
+        {
+            sum +=x % 10; //sum of digits
+            x = x / 10;
+        }
+        std::cout << "\tsum=" << sum << std::endl;
+        std::cout << "\tAverage number =" << sum / n << std::endl;
+        break;
+    case SECOND_TASK:
+        int numberDigit;
+        numberDigit = 6; //number of digits
+        std::cout << "\tEnter six digits" << std::endl;
+        std::cin >> x;
+        int chisloDigits;
+        chisloDigits = 1;
+        tmp = x; //сохранение х после подсчета количества цифр
+        while ((tmp /= 10) > 0)
+            chisloDigits++;
+        //std::cout << chisloDigits << std::endl;
+        while ((chisloDigits < numberDigit) || (chisloDigits > numberDigit))
+        {
+            chisloDigits = 1;
+            std::cout << "\tEnter not six  digits" << std::endl;
+            std::cin >> x;
+            tmp = x;
+            while ((tmp /= 10) > 0)
+                chisloDigits++;
+        }
+        int sum1, sum2;
+        sum1 = 0;
+        sum2 = 0;
+        for (int ii = 0; ii < numberDigit;ii++)
+        {
+            if (ii < numberDigit / 2) 
+             sum1 += x % 10; 
+            else
+                sum2 += x % 10;
+            x = x / 10;
+        }
+        std::cout << sum1 << std::endl;
+        std::cout << sum2 << std::endl;
+        if (sum1 == sum2)
+            std::cout << "\tAmount match" << std::endl;
+        else
+            std::cout << "\tAmounts do not match" << std::endl;
+        break;
+    case THIRD_TASK:
+        int Chislo, reversedChislo, remainder;
+        reversedChislo = 0;
+        std::cout << "Enter Number" << std::endl;
+        std::cin >> Chislo;
+        while (Chislo != 0) 
+        {
+            remainder = Chislo % 10;
+            reversedChislo = reversedChislo * 10 + remainder;
+            Chislo /= 10;
+        }
+        std::cout << "reverse number=" << reversedChislo << std::endl;
+        break;
+    case FOUR_TASK:
+        int Min, Max, cmin, cmax;
+        Min = 1;
+        Max = 50;
+        cmin = -60;
+        cmax = 90;
+        int chislo;
+        int nElement;
+        sum = 0;
+        std::cout << "\tEnter an integer in range...1...50" << std::endl;
```

Comment:
```suggestion
        std::cout << "\tEnter an integer in range... " << Min << " ..." << Max  << std::endl;
```

так ниже по коду. Это вас обезопасит от возможных ошибок если вы поменяете например значение  Min то вам не прейдется бегать по коду и смотреть где бы вы поменяли все вхождения этой переменной

Reply:
получается спасибо
std::cout << "\tEnter an integer in range..." << Min<<"--"<<Max<< std::endl;